### ISSUE TITLE
Multiprocessing pipe hangs when using en_core_web_trf

### ISSUE THREAD

**Issue by werew** [explosion/spaCy#6662]
## How to reproduce the behaviour

When running `nlp.pipe` with `n_process > 1` and using the `en_core_web_trf` model, multiprocessing seem to be stuck.
Here is a simple PoC:

```
import spacy

nlp = spacy.load("en_core_web_trf")

texts = ["Hello world" for _ in range(20)]
for doc in nlp.pipe(texts=texts, n_process=2):
    pass
```

Looks like spacy gets stuck here:
https://github.com/explosion/spaCy/blob/ace6ae435b0f1dc95f489099252b097929c9b78f/spacy/language.py#L1466

Using `en_core_web_sm` works fine.

## Info about spaCy

- **spaCy version:** 3.0.0rc2
- **Platform:** Linux-5.10.3-arch1-1-x86_64-with-glibc2.2.5
- **Python version:** 3.8.7
- **Pipelines:** en_core_web_sm (3.0.0a0), en_core_web_trf (3.0.0a0)

**Comment by svlandeg** [comment]
Thanks for the detailed report and reproducible snippet! Always much appreciated 👍 

When I run this on Windows (using `spawn`), it crashes because it can't pickle the `get_spans` function.
```
@span_getters = "spacy-transformers.strided_spans.v1"
```

I'm not sure yet whether the same problem underlies the issue you're seeing on Linux (using `fork`) but we'll have to investigate this bug either way.

**Comment by adrianeboyd** [comment]
This is mostly likely this problem with pytorch and libgomp: https://github.com/explosion/spaCy/issues/4667

See this comment (and the linked pytorch thread) for suggestions: https://github.com/explosion/spaCy/issues/4667

**Comment by adrianeboyd** [comment]
With fork it can't serialize `TransformerData` and with spawn it can't pickle the span getters. I guess for spawn we have to do something with `copy_reg`?

**Comment by martincjespersen** [comment]
Hi, I am having the same issue with spacy==3.0.3 regarding pickling the spans when running inference on a finetuned RoBERTA model with nlp.pipe(n_process > 1). Are there any news regarding this issue?

**Comment by ginward** [comment]
I am having the same issue with space version == 3.1.0 and n_process > 1. Any updates regarding this issue?

**Comment by adrianeboyd** [comment]
Hi, it looks like the issue with serializing `TransformerData` (for fork) was fixed, but I can still reproduce the pickle issue with spawn. Let me have a look...

**Comment by adrianeboyd** [comment]
I think we just need to refactor the span getter and annotation setter definitions a bit. Here's the PR: https://github.com/explosion/spacy-transformers/pull/282

**Comment by svlandeg** [comment]
This should be solved by the PR that Adriane linked - feel free to try it out by building `spacy-transformers` from source from the `master` branch. This will be available in the next release (1.0.4 or 1.1, whichever comes next)
